backport_release.md

```diff
@@ -0,0 +1,84 @@
+# Backport Release Process
+
+In situations where we need to backport commits to earlier versions of our software, we'll need to perform the release process slightly differently than a normal release. This document outlines the differences between a normal release and a backport release. It uses the same outline as the [Release Guide](release.md) and will note any change between the two types of releases.
+
+## 0. Pre-Release Checklist
+
+- Get agreement on the version number to use for the backport release. This may be an intermediate number between two preexisting releases--for example a new `0.11.2` to be added between existing `0.11.1` and `0.12.0` releases. It can also be a new latest release--so `0.13.0` in the same situation--using only some of the commits that are present in the Future Release section of the release notes.
+- Get agreement on the latest commit to use for targeting the release. A backport release will be targeted on some commit other than the latest on main. Many times the new target will be an old release, which will have a tag that can be referenced--for example `v0.11.1`.
+- Get agreement on the commits to port over for the backport release.
+
+## 0.5. Port over content for backport release
+
+#### Checkout intended target commit
+
+1. Checkout the agreed upon latest commit for targeting the release. If this is a previous release, you may checkout its tag with `git checkout v0.11.1`.
+
+#### Create backport branch
+
+1. Branch off of the target commit. For the branch name, please use the most recent major and minor versions to this commit, leaving the patch number as an `x`. This means that we would create `0.11.x`. This is necessary so that if any further backport releases are needed, we could continue to use this branch as the target. This branch is to be treated as `main` is treated in a normal release. It will be the target for our release.
+2. Protect this branch to avoid unintended commits from making their way into the release undetected.
+
+#### Port over desired commits
+
+1. Create a feature branch (this can be named whatever you'd like) and cherry-pick the desired commits.
```

Comment:
Maybe be specific and say create a feature branch off of the backport branch 0.11.x. I wonder if we can recommend a format for this branch?

Reply:
If we specify a name that should be used for the branch, say `backport_vA.B.x`, we can add a rule to our `release_notes_updated` that allows a backport PR's CI to pass if the branch name matches that format.

I'm imagining something like:
```
...
backport = '^backport_v\d+\.\d+\.x$'
regex = main, release, backport, dep_update, min_dep_update
patterns = list(map(compile, regex))
ref = "${{ github.event.pull_request.head.ref }}"
is_dev = not any(pattern.match(ref) for pattern in patterns)
...
```
Similar to the protection rule, I'm not sure whether this should happen in this PR

Reply:
Is this a code change though? I thought we update this in github settings.

Reply:
Nope this logic exists as a part of the [release_notes_updated.yml](https://github.com/alteryx/woodwork/blob/main/.github/workflows/release_notes_updated.yml) file! The protection rule is a github settings thing, though, but given that we'd need to create a protected branch to test it, I'm thinking it might be good to do that separately.

Reply:
If we change title to "Add Backport release support", this would include the yml change and doc change

Reply:
This can happen in this MR if we decide to go through with auto protecting the branch

Reply:
Added the logic in this commit (2029779) to not require a release notes for a backport PR and tested it in this PR: https://github.com/alteryx/woodwork/pull/1323